app/models/course/material/folder.rb

```diff
@@ -6,6 +6,7 @@ class Course::Material::Folder < ActiveRecord::Base
 
   has_many :materials, inverse_of: :folder, dependent: :destroy, foreign_key: :folder_id,
                        class_name: Course::Material.name, autosave: true
+  has_many :downloads, inverse_of: :folder, dependent: :destroy
```

Comment:
The jobs are technically independent of course, and we can run them outside of the app server...